.github/workflows/ci.yml

```diff
@@ -13,10 +13,71 @@ on:
       - release/**
 
 jobs:
+# add get-diff job 
+  get-diff:
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      output-contract: ${{ steps.contract.outputs.run }}
+      output-certora-contract: ${{ steps.certora-contract.outputs.run }}
+      output-scripts: ${{ steps.scripts.outputs.run }}
+      output-agents: ${{ steps.agents.outputs.run }}
+    
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get diff for contract
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-contract # via outputs
+        with:
+          PATTERNS: |
+            +(contracts)/*.sol
+
+      - id: contract
+        run: echo "::set-output name=run::true"
+        if: steps.get-diff-contract.outputs.diff
+
+      - name: Get diff for certora contract
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-certora-contract # via outputs
+        with:
+          PATTERNS: |
+            +(certora)/**/*.sol
+
+      - id: certora-contract
+        run: echo "::set-output name=run::true"
+        if: steps.get-diff-certora-contract.outputs.diff
+      
+      - name: Get diff for scripts/test
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-scripts # via outputs
+        with:
+          PATTERNS: |
+            +(scripts/test)/*.+(js|ts)
+            +(scripts/test)/**/*.+(js|ts)
+            +hardhat.config.ts
+
+      - id: scripts
+        run: echo "::set-output name=run::true"
+        if: steps.get-diff-scripts.outputs.diff
+
+      - name: Get diff for agents
+        uses: technote-space/get-diff-action@v4.0.1
+        id: get-diff-agents # via outputs
+        with:
+          PATTERNS: |
+            +(agents)/*.+(js|ts)
+            +(agents)/**/*.+(js|ts)
+            +.eslintrc.js
+
+      - id: agents
+        run: echo "::set-output name=run::true"
+        if: steps.get-diff-agents.outputs.diff
 
   generate-json-schema:
     name: .env JSON Schema Generator
     runs-on: ubuntu-latest
+    needs: get-diff
```

Comment:
I believe this job should run if there is a diff in `.env.json`. Please add the check.

Reply:
Done